coverage/PENDING_SUBMIT/amdvlk/0577-COVERAGE-constant-folding-2531-2542/cov-ucarryadd-one-and-one.amber

```diff
@@ -0,0 +1,189 @@
+#!amber
+
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+# A test for a coverage-gap found by the GraphicsFuzz project.
+
+# Short description: A fragment shader that covers specific LLVM code paths
+
+# The test passes because the shader always writes red.
+
+SHADER vertex variant_vertex_shader PASSTHROUGH
+
+# variant_fragment_shader is derived from the following GLSL:
+# #version 320 es
+# #define _int_1 _GLF_uniform_int_values[0]
+# #define _int_0 _GLF_uniform_int_values[1]
+# #define _uint_2 _GLF_uniform_uint_values[0]
+# #define _uint_1 _GLF_uniform_uint_values[1]
+#
+# precision highp float;
+# precision highp int;
+#
+# // Contents of _GLF_uniform_uint_values: [2, 1]
+# layout(set = 0, binding = 0) uniform buf0
+# {
+#     uint _GLF_uniform_uint_values[2];
+# };
+# // Contents of _GLF_uniform_int_values: [1, 0]
+# layout(set = 0, binding = 1) uniform buf1
+# {
+#     int _GLF_uniform_int_values[2];
+# };
+# layout(location = 0) out vec4 _GLF_color;
+#
+# void main()
+# {
+#     uint a = _uint_1;
+#
+#     // Always true.
+#     if(uaddCarry(1u, 1u, a) == _uint_2)
+#     {
+#         _GLF_color = vec4(_int_1, _int_0, _int_0, _int_1);
```

Comment:
Do we expect a to get set to 0 by the uaddCarry here? If so then maybe a could be used as one of the components to the _GLF_color assignment, if that wouldn't lose the coverage point?

Reply:
Seems to still trigger the coverage.